File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        result, max_time = 0, 0
        
        for i in range(len(colors)-1):
            if colors[i] == colors[i+1]:
                if neededTime[i] > neededTime[i+1]: 
                    result += neededTime[i+1]
                    neededTime[i+1] = neededTime[i]
                else:
                    result += neededTime[i]

        return result

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int n= colors.length(),gsum = 0, gmax = 0, ans = 0;
        
        for(int i = 0; i<n; i++){
            if(i>0 && colors.charAt(i) != colors.charAt(i - 1)){
                ans += gsum - gmax;
                gsum = 0;
                gmax = 0;
            }
            gsum += neededTime[i];
            gmax = Math.max(gmax, neededTime[i]);
        }
        ans += gsum - gmax;
        return ans;
    }
}